### ISSUE TITLE
Blitz should only force concurrent mode when it detects that the installed react is experimental

### ISSUE THREAD

**Issue by ryardley** [blitz-js/blitz#668]
### What is the problem?

Blitz errors out without react-experimental.

### Steps to Reproduce

1. `yarn add blitz@0.13.0 react react-dom`
1. `yarn add --dev typescript @types/react`
1. `mkdirp pages && echo 'export default () => <h1>Hello</h1>' > pages/index.tsx`
1. `yarn blitz start`

Open up the browser to http://localhost:3000

Notice errors: 

![image](https://user-images.githubusercontent.com/1256409/84466233-4b935f80-acbc-11ea-9e0c-c8cdd5b6b040.png)

### Versions

```
debug: local
debug: pkgPath: /Users/rudiyardley/projects/yatou/plain-next/node_modules/@blitzjs/cli

macOS Catalina | darwin-x64 | Node: v12.16.1

blitz: 0.13.0 (global)
blitz: 0.13.0 (local)
```

## Why is this important? 

Basically concurrent mode is not yet compatible with many existing frameworks. Your entire app must work in `<StrictMode>`

Concurrent mode should be optional for Blitz users. This is especially true for Blitz users migrating pre-existing projects

**Comment by flybayer** [comment]
💯 Great idea on auto detecting the react version!

**Comment by Daidalos117** [comment]
Any news on this? Maybe I could help with it?

**Comment by flybayer** [comment]
@Daidalos117 I'm a bit conflicted on whether to implement this or not. Because occasionally we have people auto upgrade their dependencies and accidentally switch to a non-experimental version of react. In that case, with this auto detect change, they would have weird behavior. I think they wouldn't actually have any errors, whereas currently you do.

We merged #721 after this issue was created which allows you to manually disable concurrent mode.

@ryardley what do you think?

**Comment by Daidalos117** [comment]
@flybayer Thanks, I think manually disabling experimental mode is enough. But I cannot find any documentation on it tho. Do I search bad? Maybe I can help with it and write it? Where to look how to use it correctly?
Because honestly I think this is one of the most important features. It's not possible to use Antd or even MaterialUI with experimental React (current Blitz version).

**Comment by flybayer** [comment]
@Daidalos117 yeah! Want to make a PR to add that to the docs? By the way, it should be `reactMode: 'legacy'`
